app/features/receive/receive-cashu-token.tsx

```diff
@@ -0,0 +1,97 @@
+import type { Token } from '@cashu/cashu-ts';
+import { AlertCircle } from 'lucide-react';
+import {
+  Page,
+  PageBackButton,
+  PageContent,
+  PageHeader,
+  PageHeaderTitle,
+} from '~/components/page';
+import { Button } from '~/components/ui/button';
+import { AccountSelector } from '../accounts/account-selector';
+import { LoadingScreen } from '../loading/LoadingScreen';
+import { MoneyWithConvertedAmount } from '../shared/money-with-converted-amount';
+import { useReceiveCashuToken } from './use-receive-cashu-token';
+
+type Props = {
+  token: Token;
+};
+
+export default function ReceiveToken({ token }: Props) {
+  const {
+    data: tokenData,
+    isClaiming,
+    handleClaim,
+    setReceiveAccount,
+  } = useReceiveCashuToken({
+    token,
+    cashuPubKey:
+      '038127ae202c95f4cd4ea8ba34e73618f578adf516db553a902a8589796bdc373',
+  });
+
+  if (!tokenData) {
```

Comment:
should we maybe use suspense query inside the useReceiveCashuToken so that we don't have to handle loading state?